src/key/private_key.js

```diff
@@ -238,4 +244,24 @@ class PrivateKey extends PublicKey {
   }
 }
 
+function getDefaultSubkeyType(algoName) {
+  const algo = enums.write(enums.publicKey, algoName);
+  switch (algo) {
+    case enums.publicKey.rsaEncrypt:
+    case enums.publicKey.rsaEncryptSign:
+    case enums.publicKey.rsaSign:
+    case enums.publicKey.dsa:
+    case enums.publicKey.elgamal: // impossible since it's from the primary key
+      return 'rsa';
+    case enums.publicKey.ed448:
+    case enums.publicKey.x448:
+      return 'curve448';
+    case enums.publicKey.ed25519:
+    case enums.publicKey.x25519:
+      return 'curve25519';
+    default:
+      return 'ecc';
```

Comment:
Rather than defaulting to ECC, I would prefer to spell out the options here and throw in the default case, such that we'll catch it if we forget to update it for new algorithms later